neuro_data/static_images/foundation_data_schemas.py

```diff
@@ -0,0 +1,112 @@
+import datajoint as dj
+import numpy as np
+from tqdm import tqdm
+import pandas as pd
+from foundation.fnn.data import Data
+from foundation.fnn.model import Model, Instance
+from foundation.fnn.train import Objective, Train
+from foundation.utility.resize import Resize
+from foundation.stimulus.video import FrameList
+from foundation.recording.scan import ScanUnitOrder
+from foundation.recording.trace import Trace
+from neuro_data import logger as log
+from neuro_data.static_images.data_schemas import Preprocessing, SplineCurve, FilterMixin, stimulus, fuse
+
+schema = dj.schema('neurodata_static')
+
+@schema
+class FoundationInputResponse(dj.Computed, FilterMixin):
+    definition = """ # foundation model predicted responses of all neurons to all images
+    -> Model
+    -> FrameList
+    -> Preprocessing
+    ---
+    """
+    
+    class Input(dj.Part):
+        definition = """
+        -> master
+        -> stimulus.Frame
+        ---
+        row_id              : int         # row id in the response block
+        -> stimulus.StaticImage.Image
+        """
+
+    class ResponseBlock(dj.Part):
+        definition = """
+        -> master
+        ---
+        responses           : blob@data   # response of all neurons to all trials in shape of (n_frames, n_neurons)
+        """
+
+    class ResponseKeys(dj.Part):
+        definition = """
+        -> master
+        -> fuse.Activity.Trace
+        ---
+        col_id              : int         # col id in the response block
+        """
+    
+    @property
+    def key_source(self):
+        return Model * FrameList * Preprocessing & 'preproc_id = 8'
+    
+    def make(self, key):
+        # Get stimulus parameters used for foundation model training
+        model = (Model & key).model(device="cuda")
+        model_period = (Data & key).link.compute.sampling_period
+        model_offset = (Data & key).link.compute.unit_offset
+        height, width = (Data & key).link.compute.resolution
+        resize_id = (Data & key).link.compute.resize_id
+        burnin_frames = ((Objective & (Train & (Instance() & key).link).link).link).fetch1('burnin_frames')
```

Comment:
Note: `((Objective & (Train & (Instance() & key).link).link).link)` only works for certain types of Objective.